src/doc/src/guide/build-performance.md

```diff
@@ -0,0 +1,56 @@
+# Optimizing Build Performance
+
+Compilation of Rust crates can sometimes be rather slow, due to various reasons (such as the used compilation model and the design of the Rust language and its compiler). There are various approaches that can be used to optimize build performance, which mostly fall under two categories:
+
+- Modify compiler or Cargo flags
+- Modify the source code of your crate(s)
+
+This guide focuses on the first approach.
+
+Below, you can find several methods that can be used to optimize build performance. It is important to note that their effect varies a lot based on the compiled crate, and in some cases they can actually make compilation slower. You should always measure build performance on your crate(s) to determine if a given method described here is effective for your crate.
+
+Note that some of these approaches currently require using the nightly toolchain.
+
+## Reduce amount of generated debug information
```

Comment:
For a section like this, we've been talking about making changes.  Unsure if there is an issue (and I'm generally cautious about linking to issues in documentation) but I feel like doing so could help gather more insight (e.g. the iteration vs debugging trade off) and give people a sense of "things are moving in a positive direction"